src/hotspot/cpu/aarch64/aarch64_sve_ad.m4

```diff
@@ -971,40 +985,36 @@ instruct vstoremaskL(vReg dst, vReg src, vReg tmp, immI_8 size) %{
 
 // load/store mask vector
 
-instruct vloadmask_loadV(vReg dst, vmemA mem) %{
-  predicate(UseSVE > 0 && n->as_Vector()->length_in_bytes() >= 16 &&
-            n->in(1)->bottom_type()->is_vect()->element_basic_type() == T_BOOLEAN);
-  match(Set dst (VectorLoadMask (LoadVector mem)));
-  ins_cost(2 * SVE_COST);
-  format %{ "sve_ld1b $dst, $mem\n\t"
-            "sve_neg $dst, $dst\t # load vector mask (sve)" %}
+instruct loadVMask(vReg dst, vmemA mem) %{
+  predicate(UseSVE > 0);
+  match(Set dst (LoadVectorMask mem));
+  ins_cost(SVE_COST);
+  format %{ "sve_ld1b $dst, $mem\t# load vector mask (sve)" %}
   ins_encode %{
+    // Load mask values which are boolean type, and extend them to the
+    // expected vector element type.
     FloatRegister dst_reg = as_FloatRegister($dst$$reg);
-    Assembler::SIMD_RegVariant to_vect_size =
-              elemType_to_regVariant(vector_element_basic_type(this));
+    Assembler::SIMD_RegVariant element_size =
+      elemType_to_regVariant(vector_element_basic_type(this));
     loadStoreA_predicate(C2_MacroAssembler(&cbuf), false, dst_reg, ptrue,
-                         T_BOOLEAN, to_vect_size, $mem->opcode(),
+                         T_BOOLEAN, element_size, $mem->opcode(),
                          as_Register($mem$$base), $mem$$index, $mem$$scale, $mem$$disp);
-    __ sve_neg(dst_reg, to_vect_size, ptrue, dst_reg);
   %}
   ins_pipe(pipe_slow);
 %}
 
-instruct storeV_vstoremask(vmemA mem, vReg src, vReg tmp, immI size) %{
-  predicate(UseSVE > 0 && n->as_StoreVector()->length() >= 2 &&
-            n->as_StoreVector()->vect_type()->element_basic_type() == T_BOOLEAN);
-  match(Set mem (StoreVector mem (VectorStoreMask src size)));
-  effect(TEMP tmp);
-  ins_cost(2 * SVE_COST);
-  format %{ "sve_neg $tmp, $src\n\t"
-            "sve_st1b $tmp, $mem\t # store vector mask (sve)" %}
+instruct storeVMask(vReg src, vmemA mem) %{
+  predicate(UseSVE > 0);
+  match(Set mem (StoreVectorMask mem src));
```

Comment:
Similar comment as above, existing C2 ideal node StoreVectorMaskedNode could be used here.